internal/provider/append_resource.go

```diff
@@ -224,14 +230,58 @@ func (r *AppendResource) doAppend(ctx context.Context, data *AppendResourceModel
 		zw := gzip.NewWriter(&b)
 		tw := tar.NewWriter(zw)
 		for name, f := range l.Files {
-			if err := tw.WriteHeader(&tar.Header{
-				Name: name,
-				Size: int64(len(f.Contents.ValueString())),
-				Mode: 0644,
-			}); err != nil {
-				return nil, []diag.Diagnostic{diag.NewErrorDiagnostic("Unable to write tar header", fmt.Sprintf("Unable to write tar header for %q, got error: %s", name, err))}
+			var (
+				size   int64
+				mode   int64
+				datarc io.ReadCloser
+			)
+
+			write := func(rc io.ReadCloser) error {
```

Comment:
Why is a `write` var used once below preferable to just inlining the code to write the tar header where you use it? Maybe I'm missing something.

Reply:
I think it's so we can return an error here instead of diags?